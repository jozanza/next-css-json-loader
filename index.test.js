import test from 'ava';
import loader from './index';

const content = `
.foo {
  width: 100%;
}

@keyframes bounce {
  0%, 50% { blah: .5; }
  to   { lol: 100vh; }
}

@media(min-width: 100px; max-width: 1000px) {
  .foo {
    bar: 88px;
  }
}

@font-face {
  font-family: Open Sans;
  font-style: normal;
  font-weight: 400;
  src: local('Open Sans'),
       local('OpenSans'),
       url('https://fonts.gstatic.com/s/...');
}
`;

test('es6 module with named + default exports', t => {
  const self = { context: '' };
  const output = loader.call(self, content);
  t.snapshot(output);
});
