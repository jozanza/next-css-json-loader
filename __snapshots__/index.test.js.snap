exports[`es6 module with named + default exports 1`] = `"\"export const raw = \\\"\\\\n.foo {\\\\n  width: 100%;\\\\n}\\\\n\\\\n@keyframes bounce {\\\\n  0%, 50% { blah: .5; }\\\\n  to   { lol: 100vh; }\\\\n}\\\\n\\\\n@media(min-width: 100px; max-width: 1000px) {\\\\n  .foo {\\\\n    bar: 88px;\\\\n  }\\\\n}\\\\n\\\\n@font-face {\\\\n  font-family: Open Sans;\\\\n  font-style: normal;\\\\n  font-weight: 400;\\\\n  src: local(\'Open Sans\'),\\\\n       local(\'OpenSans\'),\\\\n       url(\'https://fonts.gstatic.com/s/...\');\\\\n}\\\\n\\\"\\nexport const charset = []\\nexport const fontFace = [{\\\"fontFamily\\\":\\\"Open Sans\\\",\\\"fontStyle\\\":\\\"normal\\\",\\\"fontWeight\\\":\\\"400\\\",\\\"src\\\":\\\"local(\'Open Sans\'),\\\\n       local(\'OpenSans\'),\\\\n       url(\\\\\\\"https://fonts.gstatic.com/s/...\\\\\\\")\\\"}]\\nexport const rule = [{\\\".foo\\\":{\\\"width\\\":\\\"100%\\\"}}]\\nexport const media = [{\\\"@media(min-width: 100px; max-width: 1000px)\\\":{\\\".foo\\\":{\\\"bar\\\":\\\"88px\\\"}}}]\\nexport const keyframes = [[\\\"bounce\\\",{\\\"0%,50%\\\":{\\\"blah\\\":\\\".5\\\"},\\\"to\\\":{\\\"lol\\\":\\\"100vh\\\"}}]]\\nexport default {\\\".foo\\\":{\\\"width\\\":\\\"100%\\\"},\\\"@media(min-width: 100px; max-width: 1000px)\\\":{\\\".foo\\\":{\\\"bar\\\":\\\"88px\\\"}}}\""`;
